### ISSUE TITLE
CentOS 7.5 - Cockpit network manager dependency conflict.

### ISSUE THREAD

**Issue by PeterFalken** [cockpit-project/cockpit#9156]
When trying to update cockpit from CentOS 7.4 to CentOS 7.5, if cockpit-networkmanager was already installed, there is a dependency issue.

```
sudo yum upgrade cockpit
Transaction check error:                                                                                                                       
  file /usr/share/cockpit/networkmanager/manifest.json from install of cockpit-system-160-3.el7.centos.noarch conflicts with file from package 
cockpit-networkmanager-160-1.el7.centos.noarch                                                                                                 
                                                                                                                                               
Error Summary                                                                                                                                  
-------------                                                                                                                                  
                                                                                                                                               
[ni2@PgServer ~]$ cat /usr/share/cockpit/networkmanager/manifest.json                                                                          
{                                                                                                                                              
    "version": "160",                                                                                                                          
    "name": "network",                                                                                                                         
    "requires": {                                                                                                                              
        "cockpit": "122"                                                                                                                       
    },                                                                                                                                         
                                                                                                                                               
    "menu": {                                                                                                                                  
        "index": {                                                                                                                             
            "label": "Networking",                                                                                                             
            "order": 40                                                                                                                        
        }                                                                                                                                      
    },                                                                                                                                         
                                                                                                                                               
    "hacks": {                                                                                                                                 
        "with_networkmanager_needs_root": "yes"                                                                                                
    }                                                                                                                                          
}  
```

**Comment by martinpitt** [comment]
I confirm that cockpit-system 160-3.el7 contains networkmanager, which isn't intended. Upstream's tools/cockpit.spec doesn't do that (the merging into cockpit-system only ever happened for RHEL), and I don't see it in downstream's .spec either. So the only explanation that I have is that the package was built without the `%centos` macro defined, so that this hits:

```
%if 0%{?centos}
%define rhel 0
%endif
```

This would disable the package merging. Or maybe it built the -system package from RHEL Extras?

**Comment by PeterFalken** [comment]
Martin, thank for looking into this.
Probably related to your answer, it seems that it might have to do something with RHEL - because when I want to upgrade the cockpit package on CentOS, I get that subscription-manager and subscription-manager-rhsm are coming along as dependencies.

```
sudo yum upgrade cockpit
...
---> Package subscription-manager-rhsm.x86_64 0:1.20.11-1.el7.centos will be installed                                                                         
--> Processing Dependency: subscription-manager-rhsm-certificates = 1.20.11-1.el7.centos for package: subscription-manager-rhsm-1.20.11-1.el7.centos.x86_64    
---> Package usermode.x86_64 0:1.111-5.el7 will be installed                                                                                                   
--> Running transaction check                                                                                                                                  
---> Package libnl.x86_64 0:1.1.4-3.el7 will be installed                                                                                                      
---> Package libxml2-python.x86_64 0:2.9.1-6.el7_2.3 will be installed                                                                                         
---> Package subscription-manager-rhsm-certificates.x86_64 0:1.20.11-1.el7.centos will be installed                                                            
--> Finished Dependency Resolution                                                                                                                             
                                                                                                                                                               
Dependencies Resolved                                                                                                                                          
                                                                                                                                                               
===============================================================================================================================================================
 Package                                                   Arch                      Version                                   Repository                 Size 
===============================================================================================================================================================
Updating:                                                                                                                                                      
 cockpit                                                   x86_64                    160-3.el7.centos                          extras                     45 k 
Installing for dependencies:                                                                                                                                   
 libnl                                                     x86_64                    1.1.4-3.el7                               base                      128 k 
 libxml2-python                                            x86_64                    2.9.1-6.el7_2.3                           base                      247 k 
 python-dateutil                                           noarch                    1.5-7.el7                                 base                       85 k 
 python-dmidecode                                          x86_64                    3.12.2-2.el7                              base                       83 k 
 python-ethtool                                            x86_64                    0.8-5.el7                                 base                       33 k 
 python-inotify                                            noarch                    0.9.4-4.el7                               base                       49 k 
 subscription-manager                                      x86_64                    1.20.11-1.el7.centos                      base                      993 k 
 subscription-manager-rhsm                                 x86_64                    1.20.11-1.el7.centos                      base                      285 k 
 subscription-manager-rhsm-certificates                    x86_64                    1.20.11-1.el7.centos                      base                      195 k 
 usermode                                                  x86_64                    1.111-5.el7                               base                      193 k 
Updating for dependencies:                                                                                                                                     
 cockpit-bridge                                            x86_64                    160-3.el7.centos                          extras                    490 k 
 cockpit-system                                            noarch                    160-3.el7.centos                          extras                    1.1 M 
 cockpit-ws                                                x86_64                    160-3.el7.centos                          extras                    783 k 
                                                                                                                                                               
Transaction Summary                                                                                                                                            
===============================================================================================================================================================
Install             ( 10 Dependent packages)                                                                                                                   
Upgrade  1 Package  (+ 3 Dependent packages) 
```

**Comment by Nibbler73** [comment]
For my system it helped to remove `cockpit-networkmanager`, which now seems to be part of the `cockpit` package.

```bash
sudo yum remove cockpit-networkmanager
```

After this I could upgrade the system.

**Comment by maartenl945** [comment]
We had thr same problem with our CentOS Atomic Host build and also temporarily solved it by removing the addition of the cockpit-networkmanager.

However, I also noticed that cockpit (cockpit-system-165-3.el7.centos.noarch) on CentOS Atomic Host ( CentOS Linux 7.1805) now suddenly contains a number of menu items which were previously not there:
* Diagnostic Reports
* Kernel Dump
* SELinux
* Subscriptions

Is this perhaps also caused by the implicit inclusion of modules that were previsouyl optional ?

**Comment by martinpitt** [comment]
I'm not familiar at all how CentOS builds its packages. But this very much sounds like up to 7.4 it built packages with the `%{centos}` macro defined, while in 7.5 it stopped doing that, so that the package now gets built in RHEL mode. There cockpit-system indeed includes networkmanager, sosreport, kdump, selinux, and subscriptions.

**Comment by zul-jin** [comment]
@martinpitt I'm confident that your take on the situation is correct. The latest version of cockpit-packagekit on CentOS 7.5 requires you to set up a RHEL subscription before you can update your package which makes no sense for a CentOS system.

**Comment by PeterFalken** [comment]
@martinpitt - I had switched to the @cockpit/cockpit-preview Copr.
And that had fixed my issue so far until the lates release 178-1.
It reintroduces the RHEL subscription and few other packages (Kernel Dump, SELinux, Diagnostic Reports, Subscriptions) that aren't needed by default in a default CentOS system.

![cockpit_178](https://user-images.githubusercontent.com/2557898/45896016-17616800-bd90-11e8-9cc5-64eba90a838c.jpeg)

**Comment by martinpitt** [comment]
This was fixed for good in commit 567b68e, which is a much more robust way to prevent this problem in the future. Current COPR builds now stopped building cockpit-networkmanager and similar packages which are bundled in cockpit-system on RHEL/CentOS:

https://copr-be.cloud.fedoraproject.org/results/@cockpit/cockpit-preview/epel-7-x86_64/00802849-cockpit/
